src/main/java/com/uber/cadence/internal/replay/DecisionsHelper.java

```diff
@@ -217,6 +230,8 @@ boolean handleStartChildWorkflowExecutionFailed(HistoryEvent event) {
    */
   long requestCancelExternalWorkflowExecution(
       RequestCancelExternalWorkflowExecutionDecisionAttributes schedule) {
+    addAllMissingVersionMarker(false, Optional.empty());
```

Comment:
Can   addAllMissingVersionMarker call be just part of getNextDecisionEventId()?

Reply:
addAllMissingVersionMarker is called from all the places that leads to addDecision, not just getNextDecisionEventId.